### PR TITLE
Ensure 2020 historical data for sheet 2 is the same as 2015 historical data

### PR DESCRIPTION
**Pull request by esirK**
## Description
Ensure 2020 historical data for sheet 2 is the same as 2015 historical data

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: reports/historical/_gmmp_2015_importer.py [hunk 1]
```diff
@@ -167,7 +167,20 @@ def import_2(self, sheet_info):
...
+                if not all_new_data.get(year):
+                    all_new_data[year] = {}
+
+                for region in all_data[year]:
+                    for media in all_data[year][region]:
+                        if all_new_data.get(year).get(media):
+                            all_new_data[year][media].update(all_data[year][region][media])
+                        else:
+                            all_new_data[year][media] = all_data[year][region][media]
+        return all_new_data
```
Comment (kilemensi) [inline reports/historical/_gmmp_2015_importer.py]: QQ: Are we just removing region here?

Comment (esirK) [reply reports/historical/_gmmp_2015_importer.py]: Yes, @kilemensi. Removing region and instead we are having all countries under a certain media.
From 
Screenshot 2021-02-23 at 11 01 55

To 
Screenshot 2021-02-23 at 11 03 30

Comment (kilemensi) [reply reports/historical/_gmmp_2015_importer.py]: Then it's correct to assume the report builder knows how to group countries back to regions @esirK ?

Comment (esirK) [reply reports/historical/_gmmp_2015_importer.py]: @kilemensi  yes It does [Here](https://github.com/CodeForAfrica/gmmp/blob/master/reports/report_builder.py#L515)

**Review by kilemensi — APPROVED** [review]
🚀  @esirK 

I think we can simplify the code a bit (if I understood it correctly)
```python

for year, media_per_region in all_data.items():
    media_year_data = {}
    for region, media in media_per_region.items():
        if media not in media_year_data:
            media_year_data[media] = {}
        media_year_data[media].update(media)
 
    # we can replace the data for the year since we're done with it
    all_data[year] = media_year_data
```

**Comment by esirK** [comment]
> 🚀 @esirK
> 
> I think we can simplify the code a bit (if I understood it correctly)
> 
> ```python
> for year, media_per_region in all_data.items():
>     media_year_data = {}
>     for region, media in media_per_region.items():
>         if media not in media_year_data:
>             media_year_data[media] = {}
>         media_year_data[media].update(media)
>  
>     # we can replace the data for the year since we're done with it
>     all_data[year] = media_year_data
> ```

I've tried modifying it to work but it's not yet working. Given that we can't use a `dict` as a key to a dict, this `media_year_data[media]` and `if media not in media_year_data:` wouldn't work.

**Comment by kilemensi** [comment]
> I've tried modifying it to work but it's not yet working. Given that we can't use a `dict` as a key to a dict, this `media_year_data[media]` and `if media not in media_year_data:` wouldn't work.

Must have missed that part, so `media` is actually `{ media: { country: value }}` @esirK ?

**Comment by esirK** [comment]
> > I've tried modifying it to work but it's not yet working. Given that we can't use a `dict` as a key to a dict, this `media_year_data[media]` and `if media not in media_year_data:` wouldn't work.
> 
> Must have missed that part, so `media` is actually `{ media: { country: value }}` @esirK ?

Yep. Something like this 
```
"2010": {
        "Africa": {
          "print": {
            "benin": 0.02,
```
print being the media.
@kilemensi

**Comment by kilemensi** [comment]
> print being the media.
> @kilemensi

Cool, looks like an inner loop is/was needed @esirK 
```python
for year, country_data_per_media_per_region in all_data.items():
    year_data = {}
    for _, country_data_per_media in country_data_per_media_per_region.items():
        for media, country_data in country_data_per_media.items():
            if media not in year_data:    
                year_data[media] = {}

            year_data[media].update(country_data)
     
    # we can replace the data for the year since we're done with it
    all_data[year] = year_data
```

**Comment by esirK** [comment]
> > print being the media.
> > @kilemensi
> 
> Cool, looks like an inner loop is/was needed @esirK
> 
> ```python
> for year, country_data_per_media_per_region in all_data.items():
>     year_data = {}
>     for _, country_data_per_media in country_data_per_media_per_region.items():
>         for media, country_data in country_data_per_media.items():
>             if media not in year_data:    
>                 year_data[media] = {}
> 
>             year_data[media].update(country_data)
>      
>     # we can replace the data for the year since we're done with it
>     all_data[year] = year_data
> ```

This should work @kilemensi  but I'm investigating why the format for `all_data` is different for `2010` and `2015`. This is causing the script to fail for `2015` which has a different format.

```
{2015: {'Africa': {'benin': 0 ....
```
and thus `benin` gets treated as a print media.

**Comment by kilemensi** [comment]
> 
> ```
> {2015: {'Africa': {'benin': 0 ....
> ```
> 
> and thus `benin` gets treated as a print media.

English please @esirK : We're completely remove region i.e. `Africa` so the **_new_** `2015` format shouldn't contain `Africa` any more.

**Comment by esirK** [comment]
> English please @esirK : We're completely remove region i.e. `Africa` so the **_new_** `2015` format shouldn't contain `Africa` any more.

~~I agree @kilemensi what I'm saying is after we remove region, we expect the next key to be a media right? Check data for 2010~~
~~`{'Africa': {'print': {'benin': 0.02,`~~
~~So for 2010 `year_data[media].update(country_data)` will work. ~~
~~ For 2015 `{'Africa': {'benin': 0,`  it won't because `media` will be `benin` and `country_data` a number~~
~~The issue isn't with this script it's with how `all_data` gets generated by the `slurp_table` method.~~
~~all_data for 2015 data doesn't contain `Media`~~

**Pull request merged**